ducktape/cluster/remoteaccount.py

```diff
@@ -26,6 +27,21 @@
 from ducktape.utils.http_utils import HttpMixin
 from ducktape.utils.util import wait_until
 from ducktape.errors import DucktapeError
+import socket
+
+
+def check_ssh(method):
+    def wrapper(self, *args, **kwargs):
+        try:
+            return method(self, *args, **kwargs)
+        except (SSHException, NoValidConnectionsError, socket.error) as e:
+            if self._custom_ssh_exception_checks:
+                self._log(logging.DEBUG, "starting ssh checks:")
+                self._log(logging.DEBUG, "\n".join(repr(f) for f in self._custom_ssh_exception_checks))
+                for func in self._custom_ssh_exception_checks:
+                    func(e, self)
```

Comment:
This should also be wrapped in a try/except if we support multiple checks.

Reply:
The current assumption is that the validators will raise an exception if something is wrong.  Should this paradigm change?  Should it return some sort of result object?  We can also just try catch it but we can only raise one error in python2 (and python3 its a chain of exceptions which is a different assumption AFAIK)

Reply:
Yeah, you can only raise one exception, but you can make an exception type that wraps others. I just wasn't sure what types of checks you might do and in what order. Since this approach requires doing only the most specific checks, it can be more expensive (e.g. checking if a cloud instance is destroyed vs just doing a quick `nc -z` check first).

I'm not sure how many use cases there would be for using multiple checks other than a) fast -> slow ordering for checks and b) if you have multiple detectable failure modes. For (b), I'm not sure what that list would be.